pkg/domain/executor/workflow_executor.go

```diff
@@ -422,14 +430,21 @@ func (w *WorkflowExecutor) ExecuteActionNode(ctx context.Context, node domain.Wo
 
 	credentialIDString, ok := credentialID.(string)
 	if !ok {
-		return NodeExecutionResult{}, fmt.Errorf("credential_id is not a string")
+		err := fmt.Errorf("credential_id is not a string")
+		if node.Settings.ReturnErrorAsItem {
+			return w.createErrorAsItemResult(err, node.NodeType, node.ActionType)
+		}
+		return NodeExecutionResult{}, err
```

Comment:
Return errors directly, handle item creation outside of this function. You are duplicating.